File: ai/src/main/java/org/pingel/causality/PerfectDistribution.java

package org.pingel.causality;

import java.util.Set;

import org.pingel.bayes.Distribution;
import org.pingel.bayes.RandomVariable;


public class PerfectDistribution extends Distribution
{
    private CausalModel model;
    
    public PerfectDistribution(CausalModel model)
    {
        super(model.getRandomVariables());
        this.model = model;
    }

    
    public Set<RandomVariable> separate(RandomVariable a, RandomVariable b)
    {
        // TODO
        return null;
    }
}

File: ai/src/main/scala/org/pingel/causality/PerfectDistribution.scala

package org.pingel.causality

import org.pingel.bayes.Distribution
import org.pingel.bayes.RandomVariable

class PerfectDistribution(model: CausalModel) extends Distribution(model.getRandomVariables()) {

  def separate(a: RandomVariable, b: RandomVariable): Set[RandomVariable] = {
    // TODO
    null
  }
}
